src/main/java/recipeio/InputParser.java

```diff
@@ -26,12 +27,19 @@ public class InputParser {
 
     /**
      * Returns command entered by the user, expected to be at beginning of string.
-     * Error handling in TaskList.java
+     * Most error handling in TaskList.java, with an exception for the list command.
+     * Checks are made for any trailing characters following the list command and returns an invalid command
+     * if any of such characters are found
      *
      * @param userInput input from the user in the command line.
      * @return the command keyword. e.g. add, delete.
      */
     public static String parseCommand(String userInput) {
+        //if list command with other characters trailing, returns an invalid command
+        if (userInput.startsWith("list") && userInput.length() != 4) {
```

Comment:
i think for this we should trim the userInput in case of leading spaces. also this doesn't account for an uppercase "LIST" which should work. 

try something like if userInput.tolowercase.startsWith("list") && userInput.trim().length() != 4